elements/lisk-p2p/src/peer/base.ts

```diff
@@ -508,57 +460,61 @@ export class Peer extends EventEmitter {
 	private _resetProductivity(): void {
 		// If peer has not recently responded, reset productivity to 0
 		if (
-			this._productivity.lastResponded <
+			this.productivity.lastResponded <
 			Date.now() - DEFAULT_PRODUCTIVITY_RESET_INTERVAL
 		) {
 			this._productivity = { ...DEFAULT_PRODUCTIVITY };
 		}
 	}
-	private _updateFromProtocolPeerInfo(rawPeerInfo: unknown): void {
-		if (!this._serverNodeInfo) {
-			throw new Error('Missing server node info.');
-		}
-
-		// Sanitize and validate PeerInfo
-		const peerInfo = validatePeerInfo(
-			sanitizeIncomingPeerInfo({
-				...rawPeerInfo,
-				ipAddress: this._ipAddress,
-				wsPort: this._wsPort,
-			}),
-			this._peerConfig.maxPeerInfoSize,
+	private _updatePeerSharedState(rawSharedState: unknown): void {
+		// Validate PeerSharedState
+		const newSharedState = validateSharedState(
+			rawSharedState as P2PSharedState,
+			this.config.maxPeerInfoSize,
 		);
 
-		const result = validatePeerCompatibility(peerInfo, this._serverNodeInfo);
-
+		const result = validatePeerCompatibility(newSharedState, this.config
```

Comment:
`this.config.sharedState` is always `undefined` for inbound connections. This could cause unexpected errors.